packages/@react-spectrum/menu/src/Menu.tsx

```diff
@@ -78,13 +104,59 @@ function Menu<T extends object>(props: SpectrumMenuProps<T>, ref: DOMRef<HTMLDiv
               return menuItem;
             })}
           </div>
-        </div>
-        <div ref={scopedRef} />
+        </TrayHeaderWrapper>
+        {/* Make the portal container for submenus wide enough so that the submenu items can render as wide as they need to be */}
+        <div ref={popoverContainerRef} style={{width: '100vw', position: 'absolute', top: -5, ...leftOffset}} />
```

Comment:
This div extends the whole width of the screen so it doesn't matter if we are RTL or not. The goal is always to start the div at the farest `left` of the screen and extend to the opposite end. We extend the whole screen since menu flipping means we may need to position to either side of the button regardless of LTR or RTL